src/internal/_stepCat.js

```diff
@@ -20,7 +22,7 @@ module.exports = (function() {
   var _stepCatObject = {
     '@@transducer/init': Object,
     '@@transducer/step': function(result, input) {
-      return merge(
+      return _objectAssign(
```

Comment:
merge clones every time, which is the root cause of this issue. _objectAssign mutates the first argument.